File: Compiler/Ownership/DataFlowProfileStore.py
class DataFlowProfileStore(object):
    def __init__(self):
        self.profiles = {}

    def addProfile(self, name, profile):
        #print("addProfile: %s -> %s" % (name, profile))
        self.profiles[name] = profile

    def getProfile(self, name):
        return self.profiles[name]
    

File: Siko/src/siko/ownership/dataflowprofile/DataFlowProfileStore.rs
use super::DataFlowProfile::DataFlowProfile;
use crate::siko::qualifiedname::QualifiedName;
use std::collections::BTreeMap;

pub struct DataFlowProfileStore {
    profiles: BTreeMap<QualifiedName, DataFlowProfile>,
}

impl DataFlowProfileStore {
    pub fn new() -> DataFlowProfileStore {
        DataFlowProfileStore {
            profiles: BTreeMap::new(),
        }
    }

    pub fn addProfile(&mut self, name: QualifiedName, profile: DataFlowProfile) {
        //println!("addProfile: {} -> {:?}", name, profile);
        self.profiles.insert(name, profile);
    }

    pub fn getProfile(&self, name: &QualifiedName) -> &DataFlowProfile {
        self.profiles
            .get(name)
            .expect("data flow profile not found")
    }
}
